src/native/corehost/hostpolicy/hostpolicy_context.cpp

```diff
@@ -343,6 +343,22 @@ int hostpolicy_context_t::initialize(const hostpolicy_init_t &hostpolicy_init, c
         coreclr_properties.add(common_property::StartUpHooks, startup_hooks.c_str());
     }
 
+    // Path to dotnet corresponding to the runtime used to run the application (if it exists)
+    // Only for framework-dependent applications, as there is no corresponding dotnet for self-contained.
+#if !defined(NATIVE_LIBS_EMBEDDED)
+    if (hostpolicy_init.is_framework_dependent)
+    {
+        pal::string_t dotnet_host_path = hostpolicy_init.host_info.dotnet_root;
+        append_path(&dotnet_host_path, _X("dotnet"));
+        dotnet_host_path.append(_STRINGIFY(EXE_FILE_EXT));
```

Comment:
Can we just call `minipal_getexepath()` so the name is not hardcoded?

Reply:
The running process isn't necessarily `dotnet`. It could be an apphost and this value should still point at the `dotnet` corresponding to whatever runtime that app is running on.